Update build instructions

Angelyr opened the pull request:
-Updates SCOREC instructions to work on RHEL9
-Combines build scripts into one and moves options to environment file

Angelyr committed:
update scorec readme

Angelyr committed:
join scorec build script

Angelyr committed:
perlmutter gpu

Angelyr committed:
perlmutter cpu

Angelyr committed:
update cmake pumi-pic

Angelyr committed:
removed repeated instructions

Angelyr committed:
change full instructions

Angelyr committed:
moved gpu build

Angelyr committed:
removed unimportant changes

Angelyr committed:
whitespace change

cwsmith commented on README.md:
```diff
@@ -7,18 +7,18 @@
 The instructions in the 'install dependencies' section should be run once to get the dependencies installed.  The next section, 'Build PolyMPO', details how to do an initial build of polyMPO.  After that, the instructions in the 'develop and rebuild' section should be followed to rebuild polyMPO after making source code changes.
 
-An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For scorec users Turing GPUs are in the `cranium` and `blockade` workstations.  More info on how to set these flags for different NVIDIA architectures is below.
+An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For scorec users Turing GPUs are in the `checkers` workstations.  More info on how to set these flags for different NVIDIA architectures is below.
```
```suggestion
An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For SCOREC users, A Turing GPU is in the `checkers` workstation.  More info on how to set these flags for different NVIDIA architectures is below.
```

Angelyr committed:
update checkers comment

Co-authored-by: Cameron Smith <cwsmith@users.noreply.github.com>

cwsmith reviewed (approved):
Looks good.  ~~I have one small edit.~~  Would you like me to test the instructions?

update: Checkers has an Ampere GPU.  Sorry for the confusion with the edit 'suggestion'.

Angelyr commented:
@cwsmith Yes that would be helpful. I've tested all four builds and it should work, but maybe there is something confusing in there.

cwsmith commented:
OK. I'll give it a shot and keep you posted.  Which SCOREC system(s) did you test on?

Angelyr commented:
I've only tested on checkers

cwsmith commented on README.md:
```diff
@@ -7,18 +7,18 @@
 The instructions in the 'install dependencies' section should be run once to get the dependencies installed.  The next section, 'Build PolyMPO', details how to do an initial build of polyMPO.  After that, the instructions in the 'develop and rebuild' section should be followed to rebuild polyMPO after making source code changes.
 
-An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For scorec users Turing GPUs are in the `cranium` and `blockade` workstations.  More info on how to set these flags for different NVIDIA architectures is below.
+An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For SCOREC users, a Turing GPU is in the `checkers` workstation.  More info on how to set these flags for different NVIDIA architectures is below.
```
```suggestion
An NVIDIA GPU is required for building and running the software.  CUDA flags for NVIDIA GPUs with the Turing architecture are included below.  For SCOREC users, Turing GPUs are in the `cranium` and `blockade` workstations.  More info on how to set these flags for different NVIDIA architectures is below.
```

It looks like our changes crossed paths.  For the record, checkers has an Ampere GPU with compute capability 8.6.

Angelyr replied:
That's interesting because my build still works with this flag `-DKokkos_ARCH_TURING75=ON`

Angelyr committed:
update gpu to Ampere 8.6

cwsmith reviewed (approved):
The build and tests worked as expected on checkers.

A few notes for the polympo section:
- I think moving the clone command to the top of the section is a bit more intuitive.
- IS_TESTING -> BUILD_TESTING ?  CMake warned that IS_TESTING was ignored.
- CMake warned that CMAKE_Fortran_COMPILER was ignored for the cws/pumipicDps branch.  Is it used for other branches?

Angelyr committed:
moved clone command to top

Angelyr commented:
- I moved the clone command to the top
- I am not getting this warning. You can see the option here: https://github.com/SCOREC/polyMPO/blob/8c78cda92837bd3a3e36e1b0f37f7910c4b3563e/CMakeLists.txt#L48
- I'm not getting this warning either. But I added the `CMAKE_Fortran_COMPILER` because I was having an issue on perlmutter where the it was using multiple fortran compilers.

Angelyr committed:
fixed scorec cpu build

Angelyr committed:
missed option

cwsmith commented:
Sounds good.

I was on the wrong polympo branch.  Sorry about that.  

I'll merge this if you are all set.

onkarsahni commented:
Deleted branch for this PR.
